rbc/omnisci_backend/omnisci_buffer.py

```diff
@@ -177,6 +177,29 @@ def omnisci_buffer_getitem(x, i):
 def omnisci_buffer_ptr_setitem_(typingctx, data, index, value):
     sig = types.none(data, index, value)
 
+    eltype = data.eltype
+    nb_value = value
+
+    # [rbc issue-197] Numba promotes operations like
+    # int32(a) + int32(b) to int64
+    def truncate_or_extend(builder, value, buf_typ):
+        # buf[pos] = val
+
+        if isinstance(nb_value, types.Integer):
+            if eltype.bitwidth < nb_value.bitwidth:
+                return builder.trunc(value, buf_typ)
+            elif eltype.bitwidth > nb_value.bitwidth:
+                is_signed = nb_value.signed
+                return builder.zext(value, buf_typ) if is_signed else \
+                    builder.sext(value, buf_typ)
+            return value
+        else:  # Floating-point
```

Comment:
Could you add more tests to cover all branches of the truncate_or_extend? `int32[pos] = int32 + int64`, `int64[pos] = int32 + int64`, `int16[pos] = ...`, etc. Same for floats.

Also, is this code complete when `nb_value` is `bool`, or for the future, `str` or `array`?